code/modules/goonchat/browserassets/js/browserOutput.js

```diff
@@ -0,0 +1,910 @@
+				var newSubHeight = windowHeight - toggleHeight;
+				$(this).height(newSubHeight);
+				if (priorSubHeight > (windowHeight - toggleHeight)) {
+					$(this).addClass('scroll');
+				}
+			});
+			opts.subOptionsLoop = setInterval(function() {
+				if (!opts.suppressOptionsClose && $('#subOptions').is(':visible')) {
+					$subOptions.slideUp('fast', function() {
+						$(this).removeClass('scroll');
+						$(this).css('height', '');
+					});
+					clearInterval(opts.subOptionsLoop);
+				}
+			}, 5000); //Every 5 seconds
+		}
+	});
+
+	$('#subOptions, #toggleOptions').mouseenter(function() {
+		opts.suppressOptionsClose = true;
+	});
+
+	$('#subOptions, #toggleOptions').mouseleave(function() {
+		opts.suppressOptionsClose = false;
+	});
+
+	$('#decreaseFont').click(function(e) {
+		var fontSize = parseInt($messages.css('font-size'));
+		fontSize = fontSize - 1 + 'px';
+		$messages.css({'font-size': fontSize});
+		setCookie('fontsize', fontSize, 365);
+		internalOutput('<span class="internal boldnshit">Font size set to '+fontSize+'</span>', 'internal');
+	});
+
+	$('#increaseFont').click(function(e) {
+		var fontSize = parseInt($messages.css('font-size'));
+		fontSize = fontSize + 1 + 'px';
+		$messages.css({'font-size': fontSize});
+		setCookie('fontsize', fontSize, 365);
+		internalOutput('<span class="internal boldnshit">Font size set to '+fontSize+'</span>', 'internal');
+	});
+
+	$('#chooseFont').click(function(e) {
+		if ($('.popup .changeFont').is(':visible')) {return;}
+		var popupContent = '<div class="head">Change Font</div>' +
+			'<div id="changeFont" class="changeFont">'+
+				'<a href="#" data-font="Verdana" style="font-family: Verdana;">Verdana (Default)</a>'+
+				'<a href="#" data-font="\'Helvetica Neue\', Helvetica, Arial" style="font-family: \'Helvetica Neue\', Helvetica, Arial;">Arial / Helvetica</a>'+
+				'<a href="#" data-font="Times New Roman" style="font-family: Times New Roman;">Times New Roman</a>'+
+				'<a href="#" data-font="Georgia" style="font-family: Georgia;">Georgia</a>'+
+				'<a href="#" data-font="Courier New" style="font-family: Courier New;">Courier New</a>'+
+				'<a href="#" data-font="Lucida Console" style="font-family: Lucida Console;">Lucida Console</a>'+
+				'<a href="#" data-font="Wingdings" style="font-family: Wingdings;">Wingdings</a>'+
+				'<a href="#" data-font="Comic Sans MS" style="font-family: Comic Sans MS;">Comic Sans MS</a>'+
+			'</div>';
+		createPopup(popupContent, 200);
+	});
+
+	$('body').on('click', '#changeFont a', function(e) {
+		var font = $(this).attr('data-font');
+		$messages.css('font-family', font);
+		setCookie('fonttype', font, 365);
+	});
+
+	$('#togglePing').click(function(e) {
+		if (opts.pingDisabled) {
+			$('#ping').slideDown('fast');
+			opts.pingDisabled = false;
+		} else {
+			$('#ping').slideUp('fast');
+			opts.pingDisabled = true;
+		}
+		setCookie('pingdisabled', (opts.pingDisabled ? 'true' : 'false'), 365);
+	});
+
+	$('#saveLog').click(function(e) {
```

Comment:
@Cyberboss This shit won't work in any case. If you try `Save log` button sync ajax won't happen, but the client will experience lag, which will require restarting of the game. Async ajax would help, but in that case there is need of callback, to use css file. This bug has almost every build with goonchat, even goon themselves. And yes, problem doesn't connected with `window.XMLHtpRequest` type error.  So, you may be intrested in this decision: https://github.com/TauCetiStation/TauCetiClassic/pull/1424/commits/27d18e8e203f325a982196e8490b0f9edbdf112b
It works as well as it should. Also it fixed amazing `win.document.body.innerHTML = saved;`, where `saved` var has `<style>...</style>`, which won't work in `body` tag. Furthermore, there you can see better variant to center chat log window on the screen.